orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/adapters/TaskTasklet.groovy

```diff
@@ -173,9 +177,17 @@ class TaskTasklet implements Tasklet {
     }
 
     try {
-      def currentUser = new User(
+      // An AuthenticatedStage can override the default pipeline authentication credentials
+      def authenticatedUser = stageNavigator.findAll(stage, { Stage ancestorStage, StageBuilder stageBuilder ->
```

Comment:
Ya, that is what it's doing (traversing up the graph ... equivalent to a nearest ancestor).